### PR TITLE
Avoid unnecessary boxing of value types

### PR DESCRIPTION
**Pull request by xtqqczze**
# PR Summary

Autofix RCS1198: Avoid unnecessary boxing of value type

If the compiler added `ToString` at compile time, we wouldn't need this, see https://github.com/dotnet/roslyn/issues/44168

## PR Context

https://github.com/JosefPihrt/Roslynator/blob/master/docs/analyzers/RCS1198.md

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [NA] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [NA] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [NA] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [NA] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [NA] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [NA] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: src/Microsoft.PowerShell.Commands.Diagnostics/PdhHelper.cs [hunk 1]
```diff
@@ -545,7 +545,7 @@ public uint ConnectToDataSource(StringCollection blgFileNames)
             string doubleNullTerminated = string.Empty;
             foreach (string fileName in blgFileNames)
             {
-                doubleNullTerminated += fileName + '\0';
+                doubleNullTerminated += fileName + "\0";
             }
```
Comment (Youssef1313) [inline src/Microsoft.PowerShell.Commands.Diagnostics/PdhHelper.cs:549]: Shouldn't StringBuilder be preferred in loops instead of concat?

Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Diagnostics/PdhHelper.cs:549]: It is not hot path.

**Comment by iSazonov** [comment]
@xtqqczze I don't see a value from this change. Extra boxing is critical on hot paths but I don't see such code in the PR. I suggest to close the PR.

**Comment by xtqqczze** [comment]
@iSazonov More importantly the Roslyn compiler doesn't box on string concatenation: https://github.com/dotnet/roslyn/pull/35006

**Comment by xtqqczze** [comment]
> 
> 
> @iSazonov More importantly the Roslyn compiler doesn't box on string concatenation: [dotnet/roslyn#35006](https://github.com/dotnet/roslyn/pull/35006)

Apparently this is only for `+operator` and not `string.Concat` though:

https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxASwDYB8ACAmARgFgAoHAZgAJ9KBhSgb1MpcoAcE0A3CeaggGzUALJQCCAChwEADAG0AupQgIA5lACUzVkxKt9/GZSiUAvMrVQAdABk4AO1UwAFpQDUlAEQAPTwG5tFgBfQPZOHj5pIRxRACEpWUULdS09HVD9aSMTcyyrWgB7ewBjXgkVdVsHJ2ckL18NALTg0iCgA===

**Comment by xtqqczze** [comment]
Issue opened: https://github.com/JosefPihrt/Roslynator/issues/708

**Pull request closed without merging**